Push/PushRegistration.swift

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+public class PushRegistration: NSObject, NSCoding {
+    let endpoint: NSURL
+    let uaid: String
+    let secret: String
+    let channelID: String
+
+    public init(uaid: String, secret: String, endpoint: NSURL, channelID: String) {
+        self.uaid = uaid
+        self.secret = secret
+        self.endpoint = endpoint
+        self.channelID = channelID
+    }
+
+    @objc public convenience required init?(coder aDecoder: NSCoder) {
+        guard let uaid = aDecoder.decodeObjectForKey("uaid") as? String,
```

Comment:
I don't think it is a good idea to use `NSCoding`.  That API is still throwing uncatchable (by Swift) exceptions in case of issues when encoding or decoding. Why not use JSON instead?